This patch:

 - updates the test to run GNU make;
 - Strips all configure arguments of the form `--XXXdir=YYY` from the configure
   lines used to build the new Python tree, since these override the temporary
   --prefix being used for the test.

This patch comes from OmniOS.

--- Python-3.14.8/Tools/freeze/test/freeze.py.orig
+++ Python-3.14.8/Tools/freeze/test/freeze.py
@@ -6,6 +6,7 @@
 import sysconfig
 from test import support
 
+import re
 
 def get_python_source_dir():
     src_dir = sysconfig.get_config_var('abs_srcdir')
@@ -18,7 +19,7 @@
 TOOL_ROOT = os.path.dirname(TESTS_DIR)
 SRCDIR = get_python_source_dir()
 
-MAKE = shutil.which('make')
+MAKE = shutil.which('gmake')
 FREEZE = os.path.join(TOOL_ROOT, 'freeze.py')
 OUTDIR = os.path.join(TESTS_DIR, 'outdir')
 
@@ -143,6 +144,7 @@
     remove_opt(config_args, 'enable-optimizations')
     remove_opt(config_args, 'with-lto')
     cmd = [os.path.join(srcdir, 'configure'), *config_args]
+    cmd = [c for c in cmd if not re.search(r'^--.*dir=', c)]
     ensure_opt(cmd, 'cache-file', os.path.join(outdir, 'python-config.cache'))
     prefix = os.path.join(outdir, 'python-installation')
     ensure_opt(cmd, 'prefix', prefix)
